src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs

```diff
@@ -65,15 +67,29 @@ public static object ConvertFromJson(string input, out ErrorRecord error)
                 var dictionary = obj as JObject;
                 if (dictionary != null)
                 {
-                    obj = PopulateFromJDictionary(dictionary, out error);
+                    if (returnHashTable)
+                    {
+                        obj = PopulateHashTableFromJDictionary(dictionary, out error);
+                    }
+                    else
+                    {
+                        obj = PopulateFromJDictionary(dictionary, out error);
+                    }
                 }
                 else
                 {
                     // JArray is a collection
                     var list = obj as JArray;
                     if (list != null)
                     {
-                        obj = PopulateFromJArray(list, out error);
+                        if (returnHashTable)
+                        {
+                            obj = PopulateHashTableFromJArray(list, out error);
+                        }
+                        else
+                        {
+                            obj = PopulateFromJArray(list, out error);
+                        }
                     }
```

Comment:
Please add 'else' with Diagnostics.Assert (if the object is not JObject or JArray).

Reply:
Because of the cast to `JArray` a few lines above, `obj` cannot be something else. I initially put your suggestion in but even the compiler tells me that:
`error CS0184: The given expression is never of the provided ('JObject') type`

Reply:
I meant:
```c#
                        Dbg.Diagnostics.Assert(
                            false,
                            "Only JObject or JArray is supported.");

```